cmd/podman/login.go

```diff
@@ -60,27 +59,48 @@ func loginCmd(c *cli.Context) error {
 	if len(args) == 0 {
 		return errors.Errorf("registry must be given")
 	}
-	server := scrubServer(args[0])
+	server := registryFromFullName(scrubServer(args[0]))
 	authfile := getAuthFile(c.String("authfile"))
 
 	sc := common.GetSystemContext("", authfile, false)
 
 	// username of user logged in to server (if one exists)
-	userFromAuthFile, err := config.GetUserLoggedIn(sc, server)
+	userFromAuthFile, passFromAuthFile, err := config.GetAuthentication(sc, server)
 	if err != nil {
 		return errors.Wrapf(err, "error getting logged-in user")
 	}
-	username, password, err := getUserAndPass(c.String("username"), c.String("password"), userFromAuthFile)
+
+	ctx := getContext()
+
+	var (
+		username string
+		password string
+	)
```

Comment:
(Absolutely non-blocking nit: These declarations are unnecessary. The first `CheckAuth` can use *`FromAuthFile` directly, and the `getUserAndPass` can use `:=` for the two variables.)